packages/beacon-node/src/eth1/utils/eth1Data.ts

```diff
@@ -28,7 +28,7 @@ export async function getEth1DataForBlocks(
 
   // Collect the latest deposit of each blockNumber in a block number range
   const fromBlock = blocks[0].blockNumber;
-  const toBlock = blocks[blocks.length - 1].blockNumber;
+  const toBlock = blocks.at(-1)?.blockNumber as number;
```

Comment:
Silly questions but could we have a missing block here? Like isnt the idea of the `at` to prevent these typecast misses from bad index access?